### ISSUE TITLE
Page loader module

### ISSUE THREAD

**Issue by robert-cronin** [rjaros/kvision#45]
I think one of the things missing for KVision is a good loader/spinner module. Since a lot of the cool things KVision can do are using coroutines and ajax requests, it stands to reason that KVision needs a good loading bar. 

I saw there is `ProgressBar` which works well for our team in our forms, but we also would like a generic loader when we load our data from the backend (our app is very data intensive and makes a lot of requests).

I could just create something for our team, but would rather contribute something to KVision for others. Let me know what you think, I was thinking something like a wrapper for [nanobar.js](http://nanobar.jacoborus.codes/) or [pace.js](https://github.hubspot.com/pace/docs/welcome/)

**Comment by rjaros** [comment]
I think a module for pace.js would be very cool (I personally like the simple "Flash" theme). It could be configured automatically for all events when the module is added to the project, and reconfigured at any time with a simple API wrapper.

**Comment by robert-cronin** [comment]
I'm currently trying to add in a module for pace.js, but I am running into a little trouble when trying to use the module from Kotlin. The underlying issue is that it is a [function](https://raw.githubusercontent.com/HubSpot/pace/v0.5.3/pace.min.js) that is called when loading the module, that function then persists an object to the window and that's how Pace methods are called.

I've tried loading it in like this:
```
pl.treksoft.kvision.require("pace-js/pace.js")
```
But this spits out an error of:
```
can't resolve 'Pace' in '..../pace-js'
```
Do you have an idea of how to get the `require` to run the function straight away without those errors?

**Comment by rjaros** [comment]
Try this:
1. Use `dependency("pace-progressbar", "1.0.8")` in `build.gradle` (it seems to be the most updated fork)
2. Use:
``` 
    val pace = require("pace-progressbar").default
    val paceCss = require("pace-progressbar/themes/blue/pace-theme-flash.css")
```
to load and activate the module.

Btw, when I work with a new npm library, and I don't know how to use it, I always do this very simple test:
```
val module = require("module-name")
console.log(module)
```
Then I see in the browser console what functions and properties this module exports, and how to "dig" to the api functions described in the docs.

**Comment by robert-cronin** [comment]
awesome thanks for the pointers, its working now with `pace-progressbar`. you will hopefully see a new PR coming through sometime next week

**Comment by robert-cronin** [comment]
I am seeing a strange effect on chrome when starting PaceJs, it adds a blue border to the element:
![image](https://user-images.githubusercontent.com/8540764/58297036-c1eec680-7e08-11e9-9515-4bd94fecf325.png)
Any idea why this happens?

**Comment by rjaros** [comment]
It's probably some interaction with Bootstrap's CSS.

**Comment by robert-cronin** [comment]
yeah I think its happening because it's in focus, should be able to set border to 0.

**Comment by rjaros** [comment]
There is a "slight" problem with your new module. This line:
```
require("pace-progressbar/themes/${color.paceColorString}/pace-theme-${theme.paceThemeString}.css")
```
is broken. It makes webpack add _all_ pace themes css files to the production bundle (about 1MB of compressed "garbage"). It's because `require` function allows to use only static strings. Unfortunately there is probably no way to fix this - you need to add all themes to the bundle to be able to dynamically choose one. And this 1MB is very, very much for such small module (it makes the bigest, showcase example grow from about 3MB to about 4MB). For now I will remove this dynamic `require` and hardcode "blue flash" theme - next we can think about other solutions.

**Comment by robert-cronin** [comment]
yeah I was trying to find a way to make the theme dynamic but I didn't realize it would affect the size that much, I'm not sure how to fix that as yet

**Comment by rjaros** [comment]
I committed the following solution.

1. To use the default ("blue flash") theme we can use:
```
    override fun start(state: Map<String, Any>) {
        Pace.init()
        // other stuff ...
    }
```
2. To use any other theme we can use:
```
    override fun start(state: Map<String, Any>) {
        Pace.init(require("pace-progressbar/themes/red/pace-theme-center-radar.css"))
        // other stuff ...
    }
```
It should work fine (it will always add the default theme to the bundle, but it's only a few bytes) and should be easy enough to work with. What do you think?

**Comment by robert-cronin** [comment]
that sounds good to me, I think this issue is now resolved with version 0.0.37
